domino_data/transfer.py

```diff
@@ -0,0 +1,82 @@
+from typing import BinaryIO, Generator, Optional
+
+import io
+import shutil
+import threading
+from concurrent.futures import ThreadPoolExecutor
+
+import urllib3
+
+MAX_WORKERS = 10
+MB = 2**20
+
+
+def split_range(start: int, end: int, step: int) -> Generator[tuple[int, int], None, None]:
+    """Yield all steps from start to end as tuple of integers.
+
+    All returned block are inclusive to ensure every integer is covered from start to end.
+
+    Example:
+        start: 0, end: 10, step: 3 -> (0, 2), (3, 6), (7, 10)
```

Comment:
mmh interesting, is it normal that the first tuple in the range has one less element? like 0 -> 2 (0, 1, 2) has 3, and 3 -> 6 (3, 4, 5, 6) has 4

Reply:
it's actually wrong, I need to change with the values from the actual test. yes the latest block will not always respect the step depending on the modulo of the total size